nexus/src/app/background/tasks/webhook_deliverator.rs

```diff
@@ -0,0 +1,322 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+use crate::app::background::BackgroundTask;
+use crate::app::webhook::ReceiverClient;
+use futures::future::BoxFuture;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::DataStore;
+use nexus_db_queries::db::datastore::webhook_delivery::DeliveryAttemptState;
+pub use nexus_db_queries::db::datastore::webhook_delivery::DeliveryConfig;
+use nexus_db_queries::db::model::WebhookDeliveryAttemptResult;
+use nexus_db_queries::db::model::WebhookReceiverConfig;
+use nexus_db_queries::db::pagination::Paginator;
+use nexus_types::identity::Resource;
+use nexus_types::internal_api::background::WebhookDeliveratorStatus;
+use nexus_types::internal_api::background::WebhookDeliveryFailure;
+use nexus_types::internal_api::background::WebhookRxDeliveryStatus;
+use omicron_common::api::external::Error;
+use omicron_common::api::external::http_pagination::PaginatedBy;
+use omicron_uuid_kinds::{GenericUuid, OmicronZoneUuid, WebhookDeliveryUuid};
+use std::num::NonZeroU32;
+use std::sync::Arc;
+use tokio::task::JoinSet;
+
+// The Deliverator belongs to an elite order, a hallowed sub-category. He's got
```

Comment:
So, I'm cool if we keep this comment in, but can we also add a doc comment describing what this does? I read this file before `nexus/src/app/webhook.rs`, where this is actually described, and a snippet like:

```rust
//!  The `webhook_deliverator` task reads these delivery records and sends
//!   HTTP requests to the receiver endpoint for each delivery that is
//!   currently in flight.  The deliverator is responsible for recording the
//!   status of each *delivery attempt*.  Retries and retry backoff are
//!   the responsibility of the deliverator.
```

woulda been nice to have here

Reply:
e85868b7858ec7040b17e1338239076e404df24a added some module-level docs to both
background tasks, thanks for the suggestion!